psydac/api/feec.py

```diff
@@ -0,0 +1,145 @@
+from psydac.api.basic              import BasicDiscrete
+from psydac.feec.derivatives       import Derivative_1D, Gradient_2D, Gradient_3D
+from psydac.feec.derivatives       import ScalarCurl_2D, VectorCurl_2D, Curl_3D
+from psydac.feec.derivatives       import Divergence_2D, Divergence_3D
+from psydac.feec.global_projectors import Projector_H1, Projector_Hcurl
+from psydac.feec.global_projectors import Projector_Hdiv, Projector_L2
+from psydac.feec.pull_push         import pull_1d_h1, pull_1d_l2
+from psydac.feec.pull_push         import pull_2d_h1, pull_2d_hcurl, pull_2d_hdiv, pull_2d_l2
+from psydac.feec.pull_push         import pull_3d_h1, pull_3d_hcurl, pull_3d_hdiv, pull_3d_l2
+
+__all__ = ('DiscreteDerham',)
+
+#==============================================================================
+class DiscreteDerham(BasicDiscrete):
+    """ Represent the discrete De Rham sequence.
+    """
+    def __init__(self, mapping, *spaces):
+
+        dim           = len(spaces) - 1
+        self._dim     = dim
+        self._spaces  = spaces
+        self._mapping = mapping
+
+        if dim == 1:
+            D0 = Derivative_1D(spaces[0], spaces[1])
```

Comment:
is there a verification in the construction of D0 that the spaces form a sequence ? (same question for the other diff operators)

Reply:
no, I don't think that we verify that

Reply:
@campospinto It is not trivial to check that the spaces form a sequence, we could do that but it is quite a lot of code. 

Here we assume that they do, because these differential operators are only used by class `DiscreteDerham`, which is only created by function `discretize_derham`, which creates the appropriate spaces as part of a de Rham complex.

We could reorganize the whole process in the future. Suggestions are welcome!

Reply:
Ok ! Thanks for clarifying